integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -98,6 +103,66 @@
   }
 
   /**
+   * Changes the mode of an Alluxio file.
+   *
+   * @param path the path of the file
+   * @param mode the mode to change to
+   * @return 0 on success, a negative value on error
+   */
+  @Override
+  public int chmod(String path, @mode_t long mode) {
+    final AlluxioURI uri = mPathResolverCache.getUnchecked(path);
+
+    SetAttributeOptions options = SetAttributeOptions.defaults().setMode(new Mode((short) mode));
+    try {
+      mFileSystem.setAttribute(uri, options);
+    } catch (IOException | AlluxioException e) {
+      LOG.error("Exception on {}", path, e);
+      return -ErrorCodes.EIO();
+    }
+
+    return 0;
+  }
+
+  /**
+   * Changes the owner of an Alluxio file.
+   *
+   * This operation only works when the group mapping service is shelled based and the user is
```

Comment:
why can't we work with non-shell-based mappings? Isn't security all implemented on the master side?

Reply:
Fuse's user and group map to the ones in Unix. So shell command `chown user:group` would map to the user id and group id in Unix, which then in turn pass to this function call. However, Alluxio's security does not have user id and group id concept.

Reply:
Thanks for explaining, I get it now. Could you also include your explanation in the javadoc?